src/lib/shared/content-processor/extensions/alert.js

```diff
@@ -0,0 +1,151 @@
+  },
+  {
+    type: 'caution',
+    icon: '<svg class="octicon octicon-stop" viewBox="0 0 16 16" width="16" height="16" aria-hidden="true"><path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"></path></svg>',
+  },
+];
+
+/**
+ * Resolves the variants configuration, combining the provided variants with
+ * the default variants.
+ */
+function resolveVariants(variants) {
+  if (!variants.length) return defaultAlertVariant;
+
+  return Object.values(
+    [...defaultAlertVariant, ...variants].reduce((map, item) => {
+      map[item.type] = item;
+      return map;
+    }, {}),
+  );
+}
+
+/**
+ * Returns regex pattern to match alert syntax.
+ */
+function createSyntaxPattern(type) {
+  return `^(?:\\[!${type.toUpperCase()}])\\s*?\n*`;
+}
+
+/**
+ * Capitalizes the first letter of a string.
+ */
+function ucfirst(str) {
+  return str.slice(0, 1).toUpperCase() + str.slice(1).toLowerCase();
+}
+
+function markedAlert(options = {}) {
+  const { className = 'markdown-alert', variants = [] } = options;
+  const resolvedVariants = resolveVariants(variants);
+
+  return {
+    walkTokens(token) {
+      if (token.type !== 'blockquote') return;
+
+      const matchedVariant = resolvedVariants.find(({ type }) =>
+        new RegExp(createSyntaxPattern(type)).test(token.text),
+      );
+
+      if (!matchedVariant) return;
+
+      const {
+        type: variantType,
+        icon,
+        title = ucfirst(variantType),
+        titleClassName = `${className}-title`,
+      } = matchedVariant;
+      const typeRegexp = new RegExp(createSyntaxPattern(variantType));
+
+      Object.assign(token, {
+        type: 'alert',
+        meta: {
+          className,
+          variant: variantType,
+          icon,
+          title,
+          titleClassName,
+        },
+      });
+
+      const firstLine = token.tokens?.[0];
+      const firstLineText = firstLine.raw?.replace(typeRegexp, '').trim();
+
+      if (!firstLineText) {
+        token.tokens?.shift();
+        return;
+      }
+
+      const patternToken = firstLine.tokens[0];
+
+      Object.assign(patternToken, {
+        raw: patternToken.raw.replace(typeRegexp, ''),
+        text: patternToken.text.replace(typeRegexp, ''),
+      });
+
+      if (firstLine.tokens[1]?.type === 'br') {
+        firstLine.tokens.splice(1, 1);
+      }
```

Comment:
I know mutating is the pattern the official demos, so it's okay to me to mutate directly on `token`. However, could you avoid mutating the children of `token` by `accessing and mutating`? For readability, please consider reassgin instead of deep mutation.